app/src/main/java/at/aau/serg/logic/Authentication.kt

```diff
@@ -0,0 +1,37 @@
+package at.aau.serg.logic
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.content.Intent
+import android.widget.Toast
+import at.aau.serg.MainActivity
+import at.aau.serg.models.User
+import at.aau.serg.network.HttpClient
+import com.google.gson.Gson
+import okhttp3.Callback
+import org.json.JSONException
+import org.json.JSONObject
+
+class Authentication {
+    fun registerUser(username: String, password: String, callback: Callback): String? {
+        if(username.isEmpty() || password.isEmpty()){
+            return "Please fill all fields"
```

Comment:
As far as I can tell, these error messages are not shown to the user and it can be pretty confusing to fix an issue if all the user gets is `Registration failed`.

Reply:
The error messages are shown as a Toast to the user

Reply:
They are only shown in the `RegisterActivity` and not in the `LoginActivity`. You could also use a more specific error message, like `Username and Password cannot be empty`.